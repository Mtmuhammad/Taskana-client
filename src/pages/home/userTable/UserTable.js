// Table to display users form db
import React from "react";
import useAuth from "../../../hooks/useAuth";
import EditModal from "../editUser/EditModal";
import DeleteModal from "../deleteUser/DeleteModal";

const UserTable = ({ setSuccess, setErrMsg, setUsers, users }) => {
  const { auth } = useAuth();
  return (
    <div className="mb-5 col-12 col-lg-6 d-flex align-items-stretch">
      <div data-testid="user-table" className="table-style card w-100">
        <div className="card-body">
          <div className="d-md-flex">
            <h3 className="card-title">Team Members</h3>
          </div>
          <div>
            <div className="table-responsive mt-3">
              <table className="table stylish-table v-middle mb-0 no-wrap">
                <thead>
                  <tr>
                    <th className="border-0 text-muted fw-normal">Full Name</th>
                    <th className="border-0 text-muted fw-normal">Email</th>
                    <th className="border-0 text-muted fw-normal">Role</th>
                    {auth?.user?.isAdmin && (
                      <th className="border-0 text-muted fw-normal">Actions</th>
                    )}
                  </tr>
                </thead>
                <tbody>
                  {users?.map((user, index) => (
                    <tr key={index} className="h5">
                      <td>
                        <p className="font-weight-medium mb-0">
                          {user?.firstName} {user?.lastName}
                        </p>
                      </td>
                      <td>
                        <span className="badge bg-success px-2 py-1">
                          {user?.email}
                        </span>
                      </td>
                      <td>{user?.empRole}</td>
                      <td>
                        <div
                          className="btn-group"
                          role="group"
                          aria-label="Basic outlined example"
                        >
                          <button
                            type="button"
                            className="text-warning btn btn-outline-secondary"
                            data-bs-toggle="modal"
                            data-bs-target={`#editUser${user?.empNumber}`}
                          >
                            <i className="bx bxs-edit"></i>
                          </button>
                          <EditModal
                            key={user?.empNumber}
                            setSuccess={setSuccess}
                            setErrMsg={setErrMsg}
                            setUsers={setUsers}
                            user={user}
                          />
                          <button
                            type="button"
                            className="text-danger btn btn-outline-secondary"
                            data-bs-toggle="modal"
                            data-bs-target={`#deleteUser${user?.empNumber}`}
                          >
                            <i className="bx bx-trash"></i>
                          </button>
                          <DeleteModal
                            key={user?.empNumber}
                            setSuccess={setSuccess}
                            setErrMsg={setErrMsg}
                            setUsers={setUsers}
                            user={user}
                            users={users}
                          />
                        </div>
                      </td>
                    </tr>
                  ))}
                </tbody>
              </table>
            </div>
          </div>
        </div>
      </div>
    </div>
  );
};

export default UserTable;
