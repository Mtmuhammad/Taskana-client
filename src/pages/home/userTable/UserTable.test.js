import React from "react";
import { render, screen } from "@testing-library/react";
import UserTable from "./UserTable";
import { BrowserRouter } from "react-router-dom";

describe("smoke and snapshot tests", () => {
  test("should render without crashing", () => {
    render(<UserTable />, { wrapper: BrowserRouter });
  });

  test("should match snapshot", () => {
    const { asFragment } = render(<UserTable />, { wrapper: BrowserRouter });
    expect(asFragment()).toMatchSnapshot();
  });
});

describe("user-table", () => {
  test("should render a project table", () => {
    render(<UserTable />, { wrapper: BrowserRouter });
    const table = screen.getByTestId("user-table");
    expect(table).toBeInTheDocument();
    expect(table).toHaveClass("table-style card");
  });

  test("should render table heading", () => {
    render(<UserTable />, { wrapper: BrowserRouter });
    const heading = screen.getByText("Team Members");
    expect(heading).toBeInTheDocument();
    expect(heading).toHaveClass("card-title");
  });

  test("should render table headers", () => {
    render(<UserTable />, { wrapper: BrowserRouter });
    const headers = screen.getAllByRole("columnheader");
    expect(headers.length).toBe(4);
  });
});
