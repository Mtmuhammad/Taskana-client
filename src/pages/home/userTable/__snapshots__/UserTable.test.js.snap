// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`smoke and snapshot tests should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="mb-5 col-12 col-lg-6 d-flex align-items-stretch"
  >
    <div
      class="table-style card w-100"
      data-testid="user-table"
    >
      <div
        class="card-body"
      >
        <div
          class="d-md-flex"
        >
          <h3
            class="card-title"
          >
            Team Members
          </h3>
        </div>
        <div>
          <div
            class="table-responsive mt-3"
          >
            <table
              class="table stylish-table v-middle mb-0 no-wrap"
            >
              <thead>
                <tr>
                  <th
                    class="border-0 text-muted fw-normal"
                  >
                    Full Name
                  </th>
                  <th
                    class="border-0 text-muted fw-normal"
                  >
                    Email
                  </th>
                  <th
                    class="border-0 text-muted fw-normal"
                  >
                    Role
                  </th>
                </tr>
              </thead>
              <tbody />
            </table>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
